ci-operator/step-registry/openshift/microshift/e2e/greenboot/openshift-microshift-e2e-greenboot-commands.sh

```diff
@@ -0,0 +1,167 @@
+  sudo systemctl enable --now microshift || true
+
+  for check_script in $(find /etc/greenboot/check/ -name \*.sh | sort) ; do
+    echo Running \${check_script}...
+    local currentRC=1
+    if sudo \${check_script} ; then
+      currentRC=0
+    fi
+
+    if [ \${currentRC} != \${expectedRC} ] ; then
+      exit 1
+    fi
+  done
+}
+
+#
+# Initial check must succeed (set timeout of 60s to speed up the process)
+#
+sudo tee /etc/greenboot/greenboot.conf &>/dev/null <<EOF
+MICROSHIFT_WAIT_TIMEOUT_SEC=60
+EOF
+check_greenboot_exit_status 0
+
+#
+# User workload health
+# See https://github.com/openshift/microshift/blob/main/docs/greenboot_dev.md#user-workload-health
+#
+MANIFEST_DIR=/etc/microshift/manifests
+sudo mkdir -p \${MANIFEST_DIR}
+
+sudo tee \${MANIFEST_DIR}/busybox.yaml &>/dev/null <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: busybox
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: busybox-deployment
+spec:
+  selector:
+    matchLabels:
+      app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: BUSYBOX_IMAGE
+        command:
+          - sleep
+          - "3600"
+EOF
+
+sudo tee \${MANIFEST_DIR}/kustomization.yaml &>/dev/null <<EOF
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: busybox
+resources:
+  - busybox.yaml
+images:
+  - name: BUSYBOX_IMAGE
+    newName: registry.k8s.io/busybox
+EOF
+
+SCRIPT_FILE=/etc/greenboot/check/required.d/50_busybox_running_check.sh
+sudo curl -s https://raw.githubusercontent.com/openshift/microshift/main/docs/config/busybox_running_check.sh -o \${SCRIPT_FILE}
+sudo chmod 755 \${SCRIPT_FILE}
+
+check_greenboot_exit_status 1
+sudo rm -f \${SCRIPT_FILE}
+sudo rm -f /etc/microshift/manifests/*
+
+#
+# Service failure
+# See https://github.com/openshift/microshift/blob/main/docs/greenboot_dev.md#microshift-service-failure
+#
+sudo dnf remove -y hostname
+check_greenboot_exit_status 1
+
+sudo rm -f \${SCRIPT_FILE}
+sudo dnf install -y hostname
+
+#
+# Pod failure
+# https://github.com/openshift/microshift/blob/main/docs/greenboot_dev.md#microshift-pod-failure
+#
+sudo tee /etc/microshift/config.yaml &>/dev/null <<EOF
+network:
+  serviceNetwork:
+  - 10.66.0.0/16
+EOF
+
+check_greenboot_exit_status 1
+sudo rm -f /etc/microshift/config.yaml
+
+#
+# Last check must succeed
+#
+check_greenboot_exit_status 0
+EOFEOF
+
+chmod +x "${HOME}/greenboot_check.sh"
+
+LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
+  --quiet \
+  --project "${GOOGLE_PROJECT_ID}" \
+  --zone "${GOOGLE_COMPUTE_ZONE}" \
+  --recurse "${HOME}/greenboot_check.sh" "rhel8user@${INSTANCE_PREFIX}:~/greenboot_check.sh"
+
+LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute ssh \
+  --project "${GOOGLE_PROJECT_ID}" \
+  --zone "${GOOGLE_COMPUTE_ZONE}" \
+  "rhel8user@${INSTANCE_PREFIX}" \
+  --command "bash ~/greenboot_check.sh"
```

Comment:
If you run the script as sudo you can drop all the sudo directives in `greenboot_check.sh`